src/main.cpp

```diff
@@ -243,181 +245,241 @@ int main(const int argc, const char *argv[])
+    //             << std::endl;
+
+    //   areas.push_back(inset_state.total_inset_area());
+    //   area_errors.push_back(inset_state.max_area_error().value);
+    //   inset_state.increment_integration();
+
+    //   // if max_area_error change from previous is not lower than 5% of
+    //   // previous area error, break
+    //   if (
+    //     inset_state.max_area_error().value >
+    //     0.95 * area_errors[area_errors.size() - 2]) {
+    //     break;
+    //   }
+    // }
+
+    // // build integration vector
+    // // std::vector<double> integrations(area_errors.size());
+    // // std::iota(integrations.begin(), integrations.end(), 0);
+
+    // // plot area error and area by integration
+    // // matplot::plot(integrations, area_errors);
+    // // matplot::title(
+    // //   "Area Error vs. Integration for " + inset_state.inset_name());
+    // // matplot::xlabel("Integration");
+    // // matplot::ylabel("Area Error");
+
+    // // matplot::save("img/area_error.svg");
+
+    // // // plot area by integration
+    // // matplot::plot(integrations, areas);
+    // // matplot::title("Area vs. Integration for " +
+    // inset_state.inset_name());
+    // // matplot::xlabel("Integration");
+    // // matplot::ylabel("Area");
+
+    // // matplot::save("img/area.svg");
+
+    // if (plot_polygons) {
+
+    //   // Write PNG and PS files if requested by command-line option
+    //   std::string input_filename = inset_state.inset_name();
+    //   if (plot_grid) {
+    //     input_filename +=
+    //       "_preprocessing_grid_" +
+    //       std::to_string(inset_state.n_finished_integrations());
+    //   } else {
+    //     input_filename +=
+    //       "_preprocessing_" +
+    //       std::to_string(inset_state.n_finished_integrations());
+    //   }
+    //   std::cerr << "Writing " << input_filename << std::endl;
+    //   inset_state.write_cairo_map(input_filename, plot_grid);
+    // }
+
+    // std::cerr << "Polygon Preprocessing finished." << std::endl;
+    // time_tracker.stop("Polygon Preprocessing");
+
+    inset_state.reset_n_finished_integrations();
+
+    time_tracker.start("Integration Inset " + inset_pos);
 
     // Start map integration
     while (inset_state.n_finished_integrations() < max_integrations &&
            (inset_state.max_area_error().value > max_permitted_area_error ||
             std::abs(inset_state.area_drift() - 1.0) > 0.01)) {
+
+      std::cerr << "\nIntegration number "
+                << inset_state.n_finished_integrations() << std::endl;
+      std::cerr << "Number of Points: " << inset_state.n_points() << std::endl;
       if (qtdt_method) {
-        time_point start_delaunay_t = clock_time::now();
+        time_tracker.start("Delaunay Triangulation");
 
         // Create the Delaunay triangulation
         inset_state.create_delaunay_t();
-        time_point end_delaunay_t = clock_time::now();
-        duration_qtdt += inMilliseconds(end_delaunay_t - start_delaunay_t);
+
```

Comment:
I suggest adding the following comment: `TODO: For the vertices of a square, there are two possible Delaunay triangulations. In the current version, we lack control over the triangulation chosen by CGAL. Ideally, the triangulation should be selected that uses the shorter diagonal as a triangle edge.`

Reply:
I have added this comment on top of the function  `create_delaunay_t` in the `inset_state.cpp` file instead of `main.cpp`e, as this comment concerns the internal implementation of the method.